shell/platform/darwin/ios/ios_surface.mm

```diff
@@ -39,14 +39,20 @@ bool IsIosEmbeddedViewsPreviewEnabled() {
   }
 
 #if FLUTTER_SHELL_ENABLE_METAL
-  if ([layer.get() isKindOfClass:[CAMetalLayer class]]) {
-    return std::make_unique<IOSSurfaceMetal>(
-        fml::scoped_nsobject<CAMetalLayer>(
-            reinterpret_cast<CAMetalLayer*>([layer.get() retain])),  // Metal layer
-        std::move(context),                                          // context
-        platform_views_controller                                    // platform views controller
-    );
+#if TARGET_IPHONE_SIMULATOR
```

Comment:
Here and elsewhere, please use TARGET_OS_SIMULATOR instead of TARGET_IPHONE_SIMULATOR. The latter is deprecated by Apple in `TargetConditionals.h`.

Reply:
If we have the constant which will be 10 or 13 atm
Then the whole if is no longer needed right?